File: 01-Linkedin-Learning/-03-Programming-Foundations-Object-Oriented-design/Course_Code_&_Explanation/5-Class-Diagrams/6-static-class-variables.py
class Spaceship():

#Python doesn't use the static keyword. Instead, variables that are declared within a class, but outside of a method, are considered static or class variables
  # class variables 
  toughness = 0.8

  def __init__(self): 
    # instance variable 
    self.callSign = ""
    self._shieldStrength = 100

# When it comes to accessing instance level variables, we use the name of the object to get to them.

# in addition to static variables, we can also create static methods that exist at the class level. 

File: 01-Linkedin-Learning/-03-Programming-Foundations-Object-Oriented-design/Course_Code_&_Explanation/5-Class-Diagrams/6-static-class-variables.java

public class Spaceship {

  // instance variable 
  public string callSign;
  private int shieldStrength;

  // class variables 
  public static float toughness;

  //in addition to static variables, we can also create static methods that exist at the class level. 


  // other methods omitted
}

// When it comes to accessing instance level variables, we use the name of the object to get to them.



public class Spaceship {

  // private static variable 
  public static float toughness;

  // public static methods
  public static increaseDifficulty(float t) {
    toughness += t;
  }

  //Since static methods exist at the class level, they can only access static variables.
}